test/parallel_prefix_operations_test.dart

```diff
@@ -0,0 +1,236 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// parallel-prefix_operations.dart
+// Implementation of operations using various parallel-prefix trees.
+//
+// 2023 Sep 29
+// Author: Desmond Kirkpatrick <desmond.a.kirkpatrick@intel.com>
+
+import 'dart:math';
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/src/parallel_prefix_operations.dart';
+import 'package:test/test.dart';
+
+void testOrScan(int n, PPOrScan Function(Logic a) fn) {
+  test('or_scan_$n', () async {
+    final inp = Logic(name: 'inp', width: n);
+    final mod = fn(inp);
+    await mod.build();
+
+    int computeOrScan(int j) {
+      var result = 0;
+      var found = false;
+      for (var i = 0; i < n; ++i) {
+        if (found || ((1 << i) & j) != 0) {
+          result |= 1 << i;
+          found = true;
+        }
+      }
+      return result;
+    }
+
+    // put/expect testing
+
+    for (var j = 0; j < (1 << n); ++j) {
+      final golden = computeOrScan(j);
+      inp.put(j);
+      final result = mod.out.value.toInt();
+      //print("$j ${result} ${golden}");
+      expect(result, equals(golden));
+    }
+  });
+}
+
+void testPriorityEncoder(int n, PriorityEncoder Function(Logic a) fn) {
+  test('priority_encoder_$n', () async {
+    final inp = Logic(name: 'inp', width: n);
+    final mod = fn(inp);
+    await mod.build();
+
+    int computePriorityEncoding(int j) {
+      for (var i = 0; i < n; ++i) {
+        if (((1 << i) & j) != 0) {
+          return 1 << i;
+        }
+      }
+      return 0;
+    }
+
+    // put/expect testing
+
+    for (var j = 0; j < (1 << n); ++j) {
+      final golden = computePriorityEncoding(j);
+      inp.put(j);
+      final result = mod.out.value.toInt();
+      // print("priority_encoder: $j ${result} ${golden}");
+      expect(result, equals(golden));
+    }
+  });
+}
+
+void testAdder(int n, PPAdder Function(Logic a, Logic b) fn) {
+  test('adder_$n', () async {
+    final a = Logic(name: 'a', width: n);
+    final b = Logic(name: 'b', width: n);
+
+    final mod = fn(a, b);
+    await mod.build();
+
+    int computeAdder(int aa, int bb) => (aa + bb) & ((1 << n) - 1);
+
+    // put/expect testing
+
+    for (var aa = 0; aa < (1 << n); ++aa) {
+      for (var bb = 0; bb < (1 << n); ++bb) {
+        final golden = computeAdder(aa, bb);
+        a.put(aa);
+        b.put(bb);
+        final result = mod.out.value.toInt();
+        //print("adder: $aa $bb $result $golden");
+        expect(result, equals(golden));
+      }
+    }
+  });
+}
+
+BigInt genRandomBigInt(int inBits) {
```

Comment:
ROHD now has a way to generate random `LogicValue`s via an extension on the `Random` object

Reply:
Studying now.

Reply:
See https://intel.github.io/rohd/rohd/RandLogicValue.html